gcc/rust/resolve/rust-ast-resolve-type.cc

```diff
@@ -451,5 +469,22 @@ ResolveTypeToCanonicalPath::ResolveTypeToCanonicalPath ()
   : ResolverBase (), result (CanonicalPath::create_empty ())
 {}
 
+void
+ResolveGenericArgs::go (AST::GenericArgs &args)
+{
+  for (auto &arg : args.get_generic_args ())
+    {
+      // FIXME: Arthur: Ugly hack while waiting for disambiguation
+      if (arg.get_kind () == AST::GenericArg::Kind::Either)
```

Comment:
I think this is the right thing to do here because we want a consistent interface for all generic arg resolution to occur here so I don't thing this is ugly.

Reply:
The hack is that when faced with an amibiguous generic, we always resolve it to a type, as what was done before this patch where ambiguous generics were not handled at all. In #1358, this is where disambiguation happens and we say whether this argument is a type argument or a const value argument